torch/testing/_internal/common_utils.py

```diff
@@ -300,22 +302,40 @@ def run_tests(argv=UNITTEST_ARGS):
 
 if IS_WINDOWS:
     @contextmanager
-    def TemporaryFileName():
+    def TemporaryFileName(dir=None):
         # Ideally we would like to not have to manually delete the file, but NamedTemporaryFile
         # opens the file, and it cannot be opened multiple times in Windows. To support Windows,
         # close the file after creation and try to remove it manually
-        f = tempfile.NamedTemporaryFile(delete=False)
+        f = tempfile.NamedTemporaryFile(delete=False, dir=dir)
         try:
             f.close()
             yield f.name
         finally:
             os.unlink(f.name)
 else:
     @contextmanager  # noqa: T484
-    def TemporaryFileName():
-        with tempfile.NamedTemporaryFile() as f:
+    def TemporaryFileName(dir=None):
+        with tempfile.NamedTemporaryFile(dir=dir) as f:
             yield f.name
 
+if IS_WINDOWS:
+    @contextmanager
+    def TemporaryDirectoryName(suffix=None):
+        # On Windows the directory created by TemporaryDirectory is likely to be removed prematurely,
+        # so we first create the directory using mkdtemp and then remove it manually
+        try:
+            dir_name = tempfile.mkdtemp(suffix=suffix)
+            yield dir_name
+        finally:
+            shutil.rmtree(dir_name)
+else:
+    @contextmanager  # noqa: T484
+    def TemporaryDirectoryName(suffix=None):
+        with tempfile.TemporaryDirectory(suffix=suffix) as d:
+            yield d
+
+IS_STDIN_UTF8_ENCODING = sys.stdin.encoding.lower() == 'utf-8' and
```

Comment:
`\` missing here. BTW, I don't think this if-statement is correct. Because in `CMD`, when the display language is set to Chinese, `locale.getpreferredencoding()` returns `cp936`. The test shouldn't fail in this case.

Reply:
```suggestion
IS_STDIN_UTF8_ENCODING = sys.stdin.encoding.lower() == 'utf-8' and (sys.platform ==
                                                                    'win32' or locale.getpreferredencoding().lower() == 'utf-8')
```

Reply:
I realized that `cp936` is missing. But I was wondering, should `cp936` be considered the same as using UTF-8? Technically it's not the same but not really the same. Or should I change the variable name into something else? I can not think of a proper name now.

Reply:
No, apparently, we could just skip the check on Windows. See [PEP 528](https://www.python.org/dev/peps/pep-0528/) for more details.